api-report/counter.api.md

```diff
@@ -0,0 +1,49 @@
+## API Report File for "@fluidframework/counter"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { IChannelFactory } from '@fluidframework/datastore-definitions';
+import { IChannelStorageService } from '@fluidframework/datastore-definitions';
+import { IFluidDataStoreRuntime } from '@fluidframework/datastore-definitions';
+import { IFluidSerializer } from '@fluidframework/core-interfaces';
+import { ISequencedDocumentMessage } from '@fluidframework/protocol-definitions';
+import { ISharedObject } from '@fluidframework/shared-object-base';
+import { ISharedObjectEvents } from '@fluidframework/shared-object-base';
+import { ITree } from '@fluidframework/protocol-definitions';
+import { SharedObject } from '@fluidframework/shared-object-base';
+
+// @public
+export interface ISharedCounter extends ISharedObject<ISharedCounterEvents> {
```

Comment:
Per #6191, we should consider implementing a set() operation. The questions worth discussing here are:

1. Would adding this operation require message or event changes?
2. Is this operation critical to a known scenario?

Reply:
It would defiantly result in either new op, or extension of existing op.
same for the event - they way it's structured today is very limiting, even if we never add new APIs, I'd change event to be more inline with other dds - just reflect that there was a change, and provide before & after values - I see no good reason not to do it.

Reply:
@vladsud To make this suggestion more concrete, are you proposing a signature like this?

```ts
(event: "valueChanged", listener: (previousValue: number, newValue: number) => void): any;
```

Reply:
Tracked in #6396.